store/tikv/region_request.go

```diff
@@ -633,14 +647,22 @@ func regionErrorToLabel(e *errorpb.Error) string {
 	return "unknown"
 }
 
-func (s *RegionRequestSender) onRegionError(bo *Backoffer, ctx *RPCContext, seed *uint32, regionErr *errorpb.Error) (shouldRetry bool, err error) {
+func (s *RegionRequestSender) onRegionError(bo *Backoffer, ctx *RPCContext, seed *uint32, regionErr *errorpb.Error) (shouldRetry bool, opts []StoreSelectorOption, err error) {
 	if span := opentracing.SpanFromContext(bo.GetCtx()); span != nil && span.Tracer() != nil {
 		span1 := span.Tracer().StartSpan("tikv.onRegionError", opentracing.ChildOf(span.Context()))
 		defer span1.Finish()
 		// TODO(MyonKeminta): Make sure trace works without cloning the backoffer.
 		// bo = bo.Clone()
 		bo.SetCtx(opentracing.ContextWithSpan(bo.GetCtx(), span1))
 	}
+	opts = ctx.storeSelectorOptions
```

Comment:
Seems that we don't need to store `storeSelectorOptions` in the `RPCContext`. The exclusion of failed peer only makes sense in the loop of `RegionRequestSender.SendReqCtx`.

In the loop, you could directly pass down `opts []StoreSelectorOption`, `isStaleRead`, `lastPeerID`  to `onRegionError` and possibly return. That is, you do not need to use `RPCContext` as the "temp  accumulator of failed peers".

I am in favor of such behavior, since it keeps `RPCContext` more simple. But I am OK to the current solution anyway.

Reply:
My main concern is not adding more parameters to the `onRegionError` function since it already has `RPCContext` as a parameter. But your advice makes sense too, I will take look to figure out if there is a better way to do this.